src/Microsoft.Framework.Configuration/ConfigurationSection.cs

```diff
@@ -4,113 +4,76 @@
 using System;
 using System.Collections.Generic;
 using System.Linq;
-using Microsoft.Framework.Configuration.Internal;
 using Microsoft.Framework.Internal;
 
 namespace Microsoft.Framework.Configuration
 {
-    public class ConfigurationSection : IConfiguration
+    public class ConfigurationSection : ConfigurationBase, IConfigurationSection
     {
-        private readonly IList<IConfigurationSource> _sources = new List<IConfigurationSource>();
+        private string _key { get; }
 
-        public ConfigurationSection(IList<IConfigurationSource> sources)
+        public ConfigurationSection([NotNull] IList<IConfigurationSource> sources, [NotNull] string key)
+            : base (sources)
         {
-            _sources = sources;
+            if (string.IsNullOrEmpty(key))
+            {
+                throw new InvalidOperationException(Resources.Error_EmptyKey);
+            }
+
+            _key = key;
         }
 
-        public string this[string key]
+        public string Key
         {
             get
             {
-                return Get(key);
-            }
-            set
-            {
-                Set(key, value);
+                return _key;
             }
         }
 
-        public IEnumerable<IConfigurationSource> Sources
+        public string Value
```

Comment:
This code is so close to the indexer we should figure out reuse. But let's leave it for another PR. Can you please file it?